Facade_Engine/Query/FrameEdgeType.cs

```diff
@@ -0,0 +1,72 @@
+/*
+ * This file is part of the Buildings and Habitats object Model (BHoM)
+ * Copyright (c) 2015 - 2021, the respective contributors. All rights reserved.
+ *
+ * Each contributor holds copyright over their respective contributions.
+ * The project versioning (Git) records all such contribution source information.
+ *                                           
+ *                                                                              
+ * The BHoM is free software: you can redistribute it and/or modify         
+ * it under the terms of the GNU Lesser General Public License as published by  
+ * the Free Software Foundation, either version 3.0 of the License, or          
+ * (at your option) any later version.                                          
+ *                                                                              
+ * The BHoM is distributed in the hope that it will be useful,              
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of               
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE. See the                 
+ * GNU Lesser General Public License for more details.                          
+ *                                                                            
+ * You should have received a copy of the GNU Lesser General Public License     
+ * along with this code. If not, see <https://www.gnu.org/licenses/lgpl-3.0.html>.      
+ */
+
+using System;
+using BH.oM.Geometry;
+using BH.oM.Dimensional;
+using BH.Engine.Geometry;
+using BH.Engine.Spatial;
+using BH.oM.Facade.Elements;
+using BH.oM.Facade.SectionProperties;
+using BH.oM.Spatial.ShapeProfiles;
+using BH.oM.Analytical.Elements;
+using BH.oM.Physical.FramingProperties;
+using BH.oM.Reflection;
+using System.Collections.Generic;
+using BH.oM.Reflection.Attributes;
+using System.ComponentModel;
+using System.Linq;
+
+namespace BH.Engine.Facade
+{
+    public static partial class Query
+    {
+        /***************************************************/
+        /****              Public methods               ****/
+        /***************************************************/
+
+        [Description("Returns edge type (Sill, Head, or Jamb) from a FrameEdge and the Opening it belongs to.")]
+        [Input("frameEdge", "FrameEdge to get edge type from.")]
+        [Input("opening", "Opening the FrameEdge belongs to.")]
+        [Output("type", "FrameEdge type (Sill, Head, or Jamb).")]
+        public static string FrameEdgeType(this FrameEdge frameEdge, Opening opening)
+        {
+            if (frameEdge == null || opening == null || frameEdge.Curve == null)
+                return null;
+
+            else
+            {
+                ICurve crv = frameEdge.Curve;
+                double zVal = Math.Abs(crv.IEndDir().Z);
+                if (zVal > 0.7071)  //Check if curve is more than 45 degrees from horizontal plane
```

Comment:
Not a fan of hardcoded numbers like this - can this be a user specified tolerance? Alternatively, would need/want a Facade Engineer to confirm on this PR that this is acceptable.